packages/lesswrong/server/callbacks/rateLimits.ts

```diff
@@ -81,6 +234,105 @@ async function enforcePostRateLimit (user: DbUser) {
   }
 }
 
+
+export async function rateLimitDateWhenUserNextAbleToPost(user: DbUser): Promise<RateLimitInfo|null> {
+  // Admins and Sunshines aren't rate-limited
+  if (shouldIgnorePostRateLimit(user)) return null;
+  
+  // does the user have a moderator-assigned rate limit?
+  const modRateLimitHours = await getModRateLimitHours(user._id);
+
+  // what's the longest rate limit timeframe being evaluated?
+  const maxPostAutolimitHours = getMaxAutoLimitHours(forumSelect(autoRateLimits).posts)
+  const maxHours = Math.max(modRateLimitHours, maxPostAutolimitHours);
+
+  // fetch the posts from within the maxTimeframe
+  const postsInTimeframe = await getPostsInTimeframe(user, maxHours);
+
+  return getStrictestPostRateLimitInfo(user, postsInTimeframe, modRateLimitHours);
+}
+
+export async function rateLimitDateWhenUserNextAbleToComment(user: DbUser, postId: string | null): Promise<RateLimitInfo|null> {
+  const ignoreRateLimits = await shouldIgnoreCommentRateLimit(user, postId);
+  if (ignoreRateLimits) return null;
+
+  // does the user have a moderator-assigned rate limit?
+  const [modRateLimitHours, modPostSpecificRateLimitHours] = await Promise.all([
+    getModRateLimitHours(user._id),
+    getModPostSpecificRateLimitHours(user._id)
+  ]);
+
+  // what's the longest rate limit timeframe being evaluated?
+  const maxCommentAutolimitHours = getMaxAutoLimitHours(forumSelect(autoRateLimits).comments)
+  const maxHours = Math.max(modRateLimitHours, modPostSpecificRateLimitHours, maxCommentAutolimitHours);
+  // fetch the comments from within the maxTimeframe
+  const commentsInTimeframe = await getCommentsInTimeframe(user._id, maxHours);
+
+  return await getStrictestCommentRateLimitInfo({
+    commentsInTimeframe, 
+    user, 
+    modRateLimitHours, 
+    modPostSpecificRateLimitHours, 
+    postId
+  });
+}
+
+function getStrictestRateLimitInfo (rateLimits: Array<RateLimitInfo|null>): RateLimitInfo | null {
+  const nonNullRateLimits = rateLimits.filter(rateLimit => rateLimit !== null) as Array<RateLimitInfo>;
```

Comment:
Can be `rateLimits.filter((rateLimit) rateLimit is RateLimitInfo => rateLimit !== null)`, without a cast